### PR TITLE
enh(settings): Load from disabled users endpoint

### LINKED ISSUE #41342: Disabled users tab not working unless searching
### Steps

1. Go to user management
2. Switch to "Disabled users (1)"

### Expected

See disabled users

### Actually

If none of the first 25 users of the instance are disabled the list will be empty:
[023-11-08 14-53-41.webm](https://github.com/nextcloud/server/assets/213943/1bca0798-2ce7-453d-b0bf-ee60eda21b2d)

### Problem

It queries `…/ocs/v2.php/cloud/users/details?offset=0&limit=25&search=` without a filter for disabled users and it does not "loop"

* Ref mostlikely https://github.com/nextcloud/server/pull/39050

cc @Pytal

### PR DESCRIPTION
**Pull request by Pytal**
- Use endpoint https://github.com/nextcloud/server/pull/39756

## Summary

- Load users directly from disabled users endpoint

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by come-nc — CHANGES_REQUESTED** [review]
User count seems broken, also avatar of freshly disabled users do not load.

File: apps/settings/src/components/UserList.vue [hunk 1]
```diff
@@ -314,6 +314,10 @@ export default {
 		},
 
 		async handleScrollEnd() {
+			// Do not load more users when all have already been loaded
+			if (this.users.length % 25 !== 0) {
+				return
+			}
```
Comment (come-nc) [inline apps/settings/src/components/UserList.vue:320]: Hum, what if I load the list, enable a user from the list, and then scroll to the end? Length will be 24, no?

Comment (Pytal) [reply apps/settings/src/components/UserList.vue:320]: Nope, only `filteredUsers` would change while `users` which contains all users loaded would stay the same

**Comment by come-nc** [comment]
It works for disabled user list but for active user list it does not load the next ones when scrolling to the end it seems @Pytal

**Review by artonge — APPROVED** [review]

**Comment by come-nc** [comment]
This is still not working correctly for me for active user list.
The problem is that with LDAP backend it returns more users than the limit, because the limit is applied to each backend indivdually. This may be seen as a bug but has been this way for a long time. (and fixing that brings perf questions)

So the test that user count is a multiple of 25 does not work, in may case the first request with limit 25 and offset 0 returns 37 users (25 from ldap, the rest from DB).
What you need to test to know if there are no more users is if the number of returned users is less than the limit.

Note that for disabled users endpoint it behaves better and will slice to return only 25 users when limit is 25, but maybe it’s still safer to consider 25 users **or more** valid sizes for non-complete answer.

**Review by come-nc — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Pull request merged**